src/resdk/collection_tables.py

```diff
@@ -124,6 +126,21 @@ def __init__(
 
         self.gene_ids = []  # type: List[str]
 
+    def check_heterogeneous_collections(self):
+        """Ensure consistency among expressions."""
+        message = ""
+
+        process_slugs = {d.process.slug for d in self._data}
```

Comment:
I like using `sorted({d.process.slug for d in self._data})` when reporting iterator values.